pkg/rpc/server/server.go

```diff
@@ -1174,7 +1174,7 @@ func (s *Server) getProof(ps request.Params) (interface{}, *response.Error) {
 	skey := makeStorageKey(cs.ID, key)
 	proof, err := s.chain.GetStateModule().GetStateProof(root, skey)
 	if err != nil {
-		return nil, response.NewInternalServerError("failed to get proof", err)
+		return nil, response.NewRPCError("failed to get proof", err.Error(), err)
```

Comment:
Can be both ways, from `key is too big` to `your DB is broken`.

Reply:
Then let's consider it as an InternalServerError, just in case.